src/main/java/com/yahoo/bullet/querying/evaluators/BinaryOperations.java

```diff
@@ -0,0 +1,341 @@
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        return new TypedObject(Type.BOOLEAN, leftValue.equalTo(rightValue));
+    };
+
+    static TypedObject equalsAny(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        Type subType = leftValue.getType();
+        return new TypedObject(Type.BOOLEAN, ((List<? extends Serializable>) rightValue.getValue()).stream().anyMatch(o -> leftValue.equalTo(new TypedObject(subType, o))));
+    };
+
+    static TypedObject equalsAll(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        Type subType = leftValue.getType();
+        return new TypedObject(Type.BOOLEAN, ((List<? extends Serializable>) rightValue.getValue()).stream().allMatch(o -> leftValue.equalTo(new TypedObject(subType, o))));
+    };
+
+    static TypedObject notEquals(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        return new TypedObject(Type.BOOLEAN, !leftValue.equalTo(rightValue));
+    };
+
+    static TypedObject notEqualsAny(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        Type subType = leftValue.getType();
+        return new TypedObject(Type.BOOLEAN, ((List<? extends Serializable>) rightValue.getValue()).stream().anyMatch(o -> !leftValue.equalTo(new TypedObject(subType, o))));
+    };
+
+    static TypedObject notEqualsAll(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        Type subType = leftValue.getType();
+        return new TypedObject(Type.BOOLEAN, ((List<? extends Serializable>) rightValue.getValue()).stream().noneMatch(o -> leftValue.equalTo(new TypedObject(subType, o))));
+    };
+
+    static TypedObject greaterThan(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        return new TypedObject(Type.BOOLEAN, leftValue.compareTo(rightValue) > 0);
+    };
+
+    static TypedObject greaterThanAny(Evaluator left, Evaluator right, BulletRecord record) {
+        TypedObject leftValue = left.evaluate(record);
+        TypedObject rightValue = right.evaluate(record);
+        Type subType = leftValue.getType();
+        return new TypedObject(Type.BOOLEAN, ((List<? extends Serializable>) rightValue.getValue()).stream().anyMatch(o -> leftValue.compareTo(new TypedObject(subType, o)) > 0));
```

Comment:
Can't remember but do these work properly for vacuous truth checks? X > ANY([]) or X > ALL([])? I remember having to do stuff to handle this properly.